packages/repack/src/webpack/plugins/CodeSigningPlugin.ts

```diff
@@ -75,12 +95,23 @@ export class CodeSigningPlugin implements WebpackPlugin {
             }, {} as Record<string, string>);
 
           const json = JSON.stringify(content);
-          // Adding new asset to the compilation, so it would be automatically
-          // generated by the webpack in the output directory.
-          compilation.emitAsset(
-            this.config.outputFile,
-            new webpack.sources.RawSource(json)
-          );
+
+          if (this.config.outputPath) {
+            fs.ensureDirSync(this.config.outputPath);
+            fs.writeFileSync(
+              path.join(
+                compiler.context,
+                this.config.outputPath,
+                this.config.outputFile!
+              ),
+              json
+            );
+          } else {
+            compilation.emitAsset(
+              this.config.outputFile!,
```

Comment:
Should't the `outputFile` be marked as not-optional in the type definition? Since it's used in both branches of the if statement I'd say it's required.

Reply:
I didn't see the default value in the Plugin's constructor. All good.